js/rect.js

```diff
@@ -22,13 +22,13 @@
         return this.bottom - this.top;
       },
       set width(v) {
-        this.right = this.left + v;
+        this.right = this.left + Number(v);
```

Comment:
I'll make the change in your first comment and file an issue for generic type enforcement and changing the name of Rect to something else. How does that sound?